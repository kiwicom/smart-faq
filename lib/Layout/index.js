'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _style = require('styled-jsx/style');

var _style2 = _interopRequireDefault(_style);

var _react = require('react');

var React = _interopRequireWildcard(_react);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var style = new String('*{font-family:Roboto;}div.Layout span.name{color:blue;}');
style.__hash = '76748467';
style.__scoped = '*.jsx-4212418802{font-family:Roboto;}div.Layout.jsx-4212418802 span.name.jsx-4212418802{color:blue;}';
style.__scopedHash = '4212418802';

var Layout = function Layout() {
  return React.createElement(
    'div',
    {
      className: 'jsx-' + style.__scopedHash + ' ' + 'Layout'
    },
    'Aloha, I am a ',
    React.createElement(
      'span',
      {
        className: 'jsx-' + style.__scopedHash + ' ' + 'name'
      },
      'Layout'
    ),
    React.createElement(_style2.default, {
      styleId: style.__scopedHash,
      css: style.__scoped
    })
  );
};

exports.default = Layout;