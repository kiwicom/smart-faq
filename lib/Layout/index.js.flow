// @flow
import * as React from 'react';
import css from 'styled-jsx/css';

const style = css`
  * {
    font-family: Roboto;
  }
  div.Layout span.name {
    color: blue;
  }
`;
const Layout = () => (
  <div className="Layout">
    Aloha, I am a <span className="name">Layout</span>
    <style jsx>{style}</style>
  </div>
);

export default Layout;
