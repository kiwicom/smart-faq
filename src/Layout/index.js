// @flow
import * as React from 'react';
import css from 'styled-jsx/css';

type Props = {
  children?: React.Node,
};
const style = css`
  * {
    box-sizing: border-box;
    padding: 0;
    margin: 0;
    font-family: 'Roboto', sans-serif;
  }
  html,
  body {
    height: 100vh;
    padding: 0;
    margin: 0;
  }
  div.Layout {
    position: absolute;
    right: 0;
    min-width: 480px;
    height: 100vh;
    background-color: #ffffff;
    box-shadow: 0 4px 7px 0 rgba(0, 0, 0, 0.15);
  }
  span.inline-icon svg {
    display: inline-block;
    vertical-align: bottom;
  }
`;
const Layout = (props: Props) => (
  <div className="Layout">
    <link
      href="https://fonts.googleapis.com/css?family=Roboto"
      rel="stylesheet"
    />
    {props.children}
    <style jsx global>
      {style}
    </style>
  </div>
);

export default Layout;
