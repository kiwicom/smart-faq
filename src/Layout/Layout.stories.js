// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Layout from './index';

storiesOf('Layout', module).add('Default', () => <Layout action={action} />);
