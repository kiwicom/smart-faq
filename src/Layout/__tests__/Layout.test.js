// @flow

import * as React from 'react';
import { shallow } from 'enzyme';
import Layout from '../';

describe('Layout', () => {
  const component = shallow(<Layout />);
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  it('should render the Layout', () => {
    expect(component.html()).toMatch(/Aloha/);
  });
});
