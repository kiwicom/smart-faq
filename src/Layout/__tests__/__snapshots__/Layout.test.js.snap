// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
<div
  className="jsx-4212418802 Layout"
>
  Aloha, I am a 
  <span
    className="jsx-4212418802 name"
  >
    Layout
  </span>
  <JSXStyle
    css="*.jsx-4212418802{font-family:Roboto;}div.Layout.jsx-4212418802 span.name.jsx-4212418802{color:blue;}"
    styleId="4212418802"
  />
</div>
`;
